pkg/apis/pulumi/shared/stack_types.go

```diff
@@ -45,7 +45,7 @@ type StackSpec struct {
 	Stack string `json:"stack"`
 	// (optional) Config is the configuration for this stack, which can be optionally specified inline. If this
 	// is omitted, configuration is assumed to be checked in and taken from the source repository.
-	Config map[string]string `json:"config,omitempty"`
+	Config map[string]apiextensionsv1.JSON `json:"config,omitempty"`
```

Comment:
I'm somewhat apprehensive of switching this type even though existing strings should be compatible with this. Is it possible to maintain the `map[string]string` type, and do custom parsing/marshaling ourselves instead? I understand that this might make it slightly more difficult in terms of defining the structured config as it now has to be a scalar/string literal json representation rather than a yaml object I believe. WDYT?